js/raster_layer_editor.ts

```diff
@@ -0,0 +1,345 @@
+                visOptions.palette = this.paletteEditor?.paletteTokens() ?? [];
+            } else if (this.colorRamp === ColorRamp.Gamma) {
+                visOptions.gamma = this.gamma;
+            }
+        } else if (this.colorModel === ColorModel.RGB) {
+            visOptions.gamma = this.gamma;
+        }
+        return visOptions;
+    }
+
+    render(): TemplateResult {
+        return html`
+            <div class="vertical-flex">
+                <div class="horizontal-flex">
+                    ${this.colorModels.map((model) => {
+                        return this.renderColorModelRadio(model);
+                    })}
+                </div>
+                <div id="band-selection" class="horizontal-flex">
+                    ${this.selectedBands.map((band) => {
+                        return this.renderBandSelection(band);
+                    })}
+                </div>
+                <div class="horizontal-flex">
+                    <span class="legacy-text">Stretch:</span>
+                    ${renderSelect(this.stretchOptions, this.stretch, this.onStretchChanged)}
+                    <button
+                        class="legacy-button"
+                        @click="${this.onRefreshButtonClicked}"
+                    >
+                        <span class="material-symbols-outlined">refresh</span>
+                    </button>
+                </div>
+                <div class="horizontal-flex">
+                    <span class="legacy-text">Range:</span>
+                    <input
+                        type="text"
+                        class="legacy-text-input"
+                        id="min"
+                        name="min"
+                        .value="${this.minValue ?? "Loading..."}"
+                        @change="${this.onMinTextChanged}"
+                        ?disabled="${this.minAndMaxValuesLocked}"
+                    />
+                    <span class="legacy-text">to</span>
+                    <input
+                        type="text"
+                        class="legacy-text-input"
+                        id="max"
+                        name="max"
+                        .value="${this.maxValue ?? "Loading..."}"
+                        @change="${this.onMaxTextChanged}"
+                        ?disabled="${this.minAndMaxValuesLocked}"
+                    />
+                </div>
+                <div class="horizontal-flex">
+                    <span class="legacy-text">Opacity:</span>
+                    <input
+                        type="range"
+                        class="legacy-slider"
+                        id="opacity"
+                        name="opacity"
+                        min="0"
+                        max="1.0"
+                        step="0.01"
+                        .value=${this.opacity}
+                        @input=${this.onOpacityChanged}
+                        @change=${this.onOpacityChanged}
```

Comment:
This is probably not necessary since `input` will fire on each change whereas `change` will only fire after the input loses focus (enter/click out), but maybe there's a weird edgecase that you're trying to cover.